docs site documentation on installing airgapped Kommander (#3221)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.2/install-airgapped/index.md

---
layout: layout.pug
navigationTitle: Install air-gapped
title: Install Kommander air-gapped
menuWeight: 35
excerpt: Install Kommander in an air-gapped environment
beta: true
enterprise: false
---

This document shows how to install Kommander in an air-gapped environment. Using the air-gapped Konvoy installation documentation as the basis, this shows how to get Kommander running on top of an air-gapped Konvoy cluster.

# Naming

This document uses the following terms:
- Management cluster - Konvoy cluster that is running Kommander
- Attached cluster - Konvoy or non-Konvoy cluster that is attached to the management cluster

> * Docker registry - a registry containing all Docker images that clusters access and download during installation

Author: Can you please include this change in the followup PR you will be working on?

> * Docker registry - a registry containing all Docker images that clusters access and download during installation

Reviewer: will do

- all the prerequisites in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#before-you-begin] in case of Konvoy clusters. For attached 3rd party clusters refer to the vendor's documentation.

## Control plane nodes

Control plane nodes of Konvoy clusters should meet the minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes]. For 3rd-party clusters control plane requirements refer to the vendor specification.

## Worker nodes

Worker nodes must meet the minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#worker-nodes].

## Operating system and services for all nodes

All nodes must meet the same minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#operating-system-and-services-for-all-nodes].

## Define the inventory file

Installing air-gapped Kommander does not require any changes in the `inventory.yaml` file.

# Configure the Kubernetes cluster

> The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. Ensure all the `cluster.yaml` changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation are applied. Once this is done:

Reviewer 2: ```suggestion
The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. Ensure the `cluster.yaml` contains all the changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation. Follow this procedure:
```

1.  Ensure Kommander can use the self-hosted charts repository running on top of the Konvoy cluster. It can not connect to the default one through the public Internet.
1.  Ensure  Kommander can find and access the private Docker registry. The `registry_ip` variable is the IP address of the available private Docker registry. ${user} and ${password} are the username and password if access requires credentials.
1.  Reconfigure the Kommander controller to work in an air-gapped environment.

> The Kommander addon configuration that implements the above is as follows:

Reviewer 2: ```suggestion
Update the Kommander addon configuration like the following:
```
